lib/memjs/memjs.js

```diff
@@ -84,21 +101,20 @@ Client.create = function(serversStr, options) {
 
 // Chooses the server to talk to by hashing the given key.
 Client.prototype.server = function(key) {
-  // TODO(alevy): should use consistent hashing and/or allow swapping hashing
-  // mechanisms
-  var total = this.servers.length;
-  var origIdx = (total > 1) ? (hashCode(key) % total) : 0;
-  var idx = origIdx;
-  var serv = this.servers[idx];
-  while (serv.wakeupAt &&
-      serv.wakeupAt > Date.now()) {
-    idx = (idx + 1) % total;
-    if (idx === origIdx) {
-      return null;
-    }
-    serv = this.servers[idx];
+  return this.getServerByIndex(this.serverIndex(key));
+};
+
+Client.prototype.serverIndex = function(key) {
```

Comment:
Can you write the type signature of these functions using JSDoc syntax?
I'm confused how it works, and it would make this easier to review.
 
https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html

(Not suggesting we add a tsconfig and start trying to typecheck the JS files, but... we could and it would be minimally invasive.)

Reply:
aded jsdocs